### PR TITLE
Sync with florian's fork

### PR DESCRIPTION
**Pull request by florianschanda**
Sync with Florian's fork

Implements:
* bmw-software-engineering/trlc#85
* bmw-software-engineering/trlc#62
* bmw-software-engineering/trlc#53

Also fixes up some documentation and adds design docs for a new feature.

### CODE REVIEW

**Review by christophkloeffel — CHANGES_REQUESTED** [review]
also remove https://github.com/florianschanda/trlc/blob/43fbfa3c7161d231479d3e845beb19c85cb89eeb/Makefile#L97C23-L97C59 because it is not needed anymore

File: documentation/architecture.md [hunk 1]
```diff
@@ -184,9 +184,8 @@ the `Source_Manager`.
   message.
 
 * During this we build a dependency graph, for each package we create
-  three nodes in a graph: a rsl node, a check node, and a trlc
-  node. Initially we link the check to the rsl; and the trlc to both
-  the check and rsl.
+  two nodes in a graph: a rsl node and a trlc node. Initially we the
+  the trlc to both the rsl.
```
[the remark this answers is not in the document]
Comment (florianschanda) [reply documentation/architecture.md]: There were a few more issues with this commit (tutorials and architecture doc) that I've fixed as well.

**Review by christophkloeffel — APPROVED** [review]

**Comment by christophkloeffel** [comment]
thx a lot

**Comment by florianschanda** [comment]
@christophkloeffel for these kind of bundled commits, and only these, please use rebase merging so my commit history doesn't get lost.

We want squash merging for everything else, but for these we need to use rebase merging. For example the individual Closes #issuenumber magic text disappears, so the issues need to be manually closed.

Of course separate PRs can be used to work around this, but it's often not reasonable if I have to create multiple commits that build on each other.

**Comment by christophkloeffel** [comment]
@florianschanda please open another PR with the same content and I will do that

**Pull request merged**